test/concurrency/lock_manager_test.cpp

```diff
@@ -0,0 +1,133 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// lock_manager_test.cpp
+//
+// Identification: test/concurrency/lock_manager_test.cpp
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "concurrency/lock_manager.h"
+#include "common/harness.h"
+
+namespace peloton {
+namespace test {
+
+//===--------------------------------------------------------------------===//
+// Lock Manager functionality tests
+//===--------------------------------------------------------------------===//
+
+class LockManagerTests : public PelotonTest {};
+
+/**
+ * @brief Test basic functionality of lock manager
+ */
+TEST_F(LockManagerTests, FunctionalityTest) {
+  // Initialize lock
+  concurrency::LockManager *lm = concurrency::LockManager::GetInstance();
+  bool result0 = lm->InitLock((oid_t)1, concurrency::LockManager::RW_LOCK);
+  // Try to lock shared
+  bool result1 = lm->LockShared((oid_t)1);
+  // Try to unlock
+  bool result2 = lm->UnlockShared((oid_t)1);
+  // Try to remove lock
+  bool result3 = lm->RemoveLock((oid_t)1);
+  EXPECT_TRUE(result0);
+  EXPECT_TRUE(result1);
+  EXPECT_TRUE(result2);
+  EXPECT_TRUE(result3);
+}
+
+/**
+ * @brief Test shared lock behaviour
+ */
+TEST_F(LockManagerTests, LockSharedTest) {
+  // Initialize lock
+  concurrency::LockManager *lm = concurrency::LockManager::GetInstance();
+  bool result0 = lm->InitLock((oid_t)1, concurrency::LockManager::RW_LOCK);
```

Comment:
Well I don't think this could be a problem. The point to add comment here is, this feature is newly added, and I think it's necessary to remind other developers of its existence.

Reply:
IMO, adding some block comments about the test functionality is enough...  We don't need to comment every single statement...